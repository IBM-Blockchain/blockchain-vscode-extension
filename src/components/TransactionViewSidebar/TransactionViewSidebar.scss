@import '@carbon/colors/scss/colors';

.sidebar-container {
    border-right: 1px solid $carbon--gray-20;
}
