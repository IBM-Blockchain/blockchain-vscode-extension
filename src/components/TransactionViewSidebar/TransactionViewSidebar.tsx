import React from 'react';
import './TransactionViewSidebar.scss';
import SidebarPanel from '../TransactionViewSidebarPanel/TransactionViewSidebarPanel';

function TransactionViewSidebar(): any {

        return (
            <div className="sidebar-container" id="sidebar-container">
                <SidebarPanel panelType={'buttons'}/>
                <SidebarPanel panelType={'filters'}/>
                <SidebarPanel panelType={'log'}/>
            </div>
        );
}

export default TransactionViewSidebar;
