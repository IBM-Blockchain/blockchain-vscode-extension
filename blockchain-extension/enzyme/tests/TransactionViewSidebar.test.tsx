import React from 'react';
import renderer from 'react-test-renderer';
import TransactionViewSidebar from '../../src/components/TransactionViewSidebar/TransactionViewSidebar';
import chai from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';
chai.should();
chai.use(sinonChai);

describe('TransactionViewSidebar component', () => {

    let mySandBox: sinon.SinonSandbox;
    
    beforeEach(async () => {
        mySandBox = sinon.createSandbox();
    });

    afterEach(async () => {
        mySandBox.restore();
    });

    it('should render the expected snapshot', async () => {
        const component: any = renderer
            .create(<TransactionViewSidebar/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

});
