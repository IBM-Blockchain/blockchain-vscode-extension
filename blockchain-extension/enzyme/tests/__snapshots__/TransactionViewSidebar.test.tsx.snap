// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionViewSidebar component should render the expected snapshot 1`] = `
<div
  className="sidebar-container"
  id="sidebar-container"
>
  <div
    className="panel-container"
  >
    <button
      className="bx--btn bx--btn--field bx--btn--primary"
      disabled={false}
      id="create-button"
      onClick={[Function]}
      tabIndex={0}
      type="button"
    >
      Create a new transaction
    </button>
    <button
      className="bx--btn bx--btn--field bx--btn--primary"
      disabled={false}
      id="import-button"
      onClick={[Function]}
      tabIndex={0}
      type="button"
    >
      Import existing transaction
    </button>
  </div>
  <div
    className="panel-container"
  >
    <h5>
      Filters
    </h5>
  </div>
  <div
    className="panel-container"
  >
    <h5>
      Transactions
    </h5>
  </div>
</div>
`;
